feat(command-env): add migrate subcommand

ashalfarhan opened the pull request:
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes #3575

This PR will add the ability to transfer environment variables from one site to another.
If you want to send env vars from one site to another you can run this subcommand.
Given 2 arguments, the basic usage is:
- If both arguments are provided, then the first argument will be the sender, and the second one is the destination
- If only one argument is provided, then that will be the destination, and the sender should be the current directory site, otherwise, it will exit with a helpful error message
- If there's no env from `siteIdA`, then it will exit because nothing to transfer

So users can use this command without opening their browser to copy and paste the env vars from one to another.

### Notes:
- Currently this just only works with site id.

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

![koala5](https://user-images.githubusercontent.com/70800415/140144742-6bbdba96-08bf-4c7f-ad17-0abbabbe1a74.jpg)

erezrokah reviewed (commented):
@ashalfarhan, sorry for the very later review.

I changed the command a bit to use options instead of arguments.
It makes the purpose of each site id more explicit.
I also added error handling to handle when the CLI can't get the site from the id.
Finally, I think `transfer` can be confusing for this command, as it doesn't remove the variables from the `from` site (I should have figured this out sooner).

We could try calling it `env:migrate` or `env:clone`, what do you think?

ashalfarhan commented:
> @ashalfarhan, sorry for the very later review.
> 
> I changed the command a bit to use options instead of arguments. It makes the purpose of each site id more explicit. I also added error handling to handle when the CLI can't get the site from the id. Finally, I think `transfer` can be confusing for this command, as it doesn't remove the variables from the `from` site (I should have figured this out sooner).
> 
> We could try calling it `env:migrate` or `env:clone`, what do you think?

Yeah, I think it's better to use options instead of arguments.
And sorry to make you work for the error handling 🙏, because I am still not sure should I just log the error or I should do something, so I just let the error be thrown.

Btw `env:migrate` is more relevant for this feature I think.

erezrokah reviewed (approved):
🎸
